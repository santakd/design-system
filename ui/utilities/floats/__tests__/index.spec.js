/* eslint-env jest */
import React from 'react';
import createHelpers from '../../../../jest.helpers';
const { matchesMarkup } = createHelpers(__dirname);

describe('renders float utilities', () => {
  it('renders float left', () =>
    matchesMarkup(
      <div className="slds-clearfix">
        <div className="slds-float_left">
          <p>I’m floooaaaating</p>
        </div>
      </div>
    ));

  it('renders float right', () =>
    matchesMarkup(
      <div className="slds-clearfix">
        <div className="slds-float_right">
          <p>I’m floooaaaating</p>
        </div>
      </div>
    ));

  it('renders float none', () =>
    matchesMarkup(
      <div className="slds-float_none">
        <p>I’m not floooaaaating</p>
      </div>
    ));

  it('renders floats with a clearfix', () =>
    matchesMarkup(
      <div className="slds-clearfix">
        <div className="slds-float_left">
          <p>I’m floooaaaating</p>
        </div>
        <div className="slds-float_right">
          <p>I’m floooaaaating</p>
        </div>
      </div>
    ));
});
