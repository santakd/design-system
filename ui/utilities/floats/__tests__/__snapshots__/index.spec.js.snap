// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders float utilities renders float left 1`] = `
<div
  className="slds-clearfix"
>
  <div
    className="slds-float_left"
  >
    <p>
      I’m floooaaaating
    </p>
  </div>
</div>
`;

exports[`renders float utilities renders float none 1`] = `
<div
  className="slds-float_none"
>
  <p>
    I’m not floooaaaating
  </p>
</div>
`;

exports[`renders float utilities renders float right 1`] = `
<div
  className="slds-clearfix"
>
  <div
    className="slds-float_right"
  >
    <p>
      I’m floooaaaating
    </p>
  </div>
</div>
`;

exports[`renders float utilities renders floats with a clearfix 1`] = `
<div
  className="slds-clearfix"
>
  <div
    className="slds-float_left"
  >
    <p>
      I’m floooaaaating
    </p>
  </div>
  <div
    className="slds-float_right"
  >
    <p>
      I’m floooaaaating
    </p>
  </div>
</div>
`;
